agent/command/git.go

```diff
@@ -870,15 +871,22 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		}
 	}
 
-	// Clone the project's modules.
+	g, gCtx := errgroup.WithContext(ctx)
+	g.SetLimit(10)
+
+	// Clone the project's modules in goroutines.
 	for _, moduleName := range conf.BuildVariant.Modules {
-		if err := ctx.Err(); err != nil {
-			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
-		}
-		err = c.fetchModuleSource(ctx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName)
-		if err != nil {
-			return errors.Wrapf(err, "fetching module source '%s'", moduleName)
-		}
+		moduleName := moduleName
+		g.Go(func() error {
+			if err := gCtx.Err(); err != nil {
+				return nil
+			}
+			return errors.Wrapf(c.fetchModuleSource(gCtx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName), "fetching module source '%s'", moduleName)
```

Comment:
`gCtx` is safe to use in the closure because contexts are safe for access from multiple goroutines, but what about everything else? Is there a risk of a race condition here? We are creating multiple goroutines with access to some things that may not be concurrency-safe. Does `fetchModuleSource()` modify any of those objects?

Reply:
Yes, I just checked throughout and it doesn't have any non-current safe. The only thing I see (and can notice in the logs of the tasks) is that the clone output is jumbled with eachother. I don't think that's much of a risk considering how stable git clone is- and often if there is an error it is just a permissions error

Reply:
No modifications though to be clear ^

Reply:
I ran multiple tests to make sure, and it does have a map update for some module paths. I'm updating it now to not do this/remove concurrent updates error

Reply:
Thanks for checking!